### ISSUE TITLE
Robust Descriptors

### ISSUE THREAD

**Issue by mmmfarrell** [rpautrat/SuperPoint#44]
I read a few closed issues (#22) that you mentioned that you still hadn't gotten the descriptors to be very robust and therefore you can't really match features between two different images. I'm curious if you, or anyone, has since gotten the descriptors to work robustly? I think this is the best part of the original SuperPoint implementation by MagicLeap. From my experience, descriptor matching across images is super robust on their implementation.

If you, or anyone else has gotten it to work, could you please explain your methods a little bit so that I can replicate them?

Thanks!!!

**Comment by rpautrat** [comment]
Hi,

Indeed I had an issue with the descriptors for a long time, but it is fixed now and the master branch is now able to produce robust descriptors (see the results in the ReadMe).

The reason why it didn't work was that there was still 2-3 small bugs in the computation of the descriptors loss. And tuning the hyperparameters described in the original article helps to improve the score as well (like the parameter balancing the detector and descriptor losses, the factor compensating the lower number of positive matches compared to the negative ones, etc).

**Comment by mmmfarrell** [comment]
@rpautrat  Awesome, thanks for the quick response and great work with the implementation! I will put in time to get it trained and running on my computer in the next few days. Do you not intend to upload pretrained weights again? I saw that in the last [commit](https://github.com/rpautrat/SuperPoint/commit/cbcc3d1485736721e5c0388884ea99faa6d6cce5) you removed the old weights that didn't work great, but now that it is working, do you intend to upload new ones? I think it might be very useful for people like myself who want to evaluate quickly if the network will work well enough for my specific application or not. If not, no worries, I totally get it! Thanks for the work!

**Comment by rpautrat** [comment]
Yes actually I intend to reupload pretrained weights with the correct version of SuperPoint, but for some obscure reason the weights and graph that I exported are giving weird detections (whereas the trained network works perfectly well). So I just need to fix this issue and then I should be able to upload the weight.
However I have quite a lot of other things to do right now, so it is on my todo list but don't expect to see it too soon :)

**Comment by mmmfarrell** [comment]
Haha no worries, I totally get it. Thanks man, I appreciate your work!
